afterpay/src/main/kotlin/com/afterpay/android/view/AfterpayBadge.kt

```diff
@@ -0,0 +1,62 @@
+package com.afterpay.android.view
+
+import android.content.Context
+import android.util.AttributeSet
+import android.util.TypedValue
+import android.view.Gravity
+import android.view.View
+import android.widget.FrameLayout
+import android.widget.ImageView
+import com.afterpay.android.R
+
+class AfterpayBadge(context: Context, attrs: AttributeSet?) : FrameLayout(context, attrs) {
+    constructor(context: Context) : this(context, attrs = null)
+
+    var colorScheme: AfterpayColorScheme = AfterpayColorScheme.DEFAULT
+        set(value) {
+            field = value
+            applyColorScheme()
+            invalidate()
+            requestLayout()
+        }
+
+    private val badgeView = ImageView(context).apply {
+        setImageResource(colorScheme.badgeDrawable)
+        adjustViewBounds = true
+        scaleType = ImageView.ScaleType.FIT_CENTER
+        importantForAccessibility = View.IMPORTANT_FOR_ACCESSIBILITY_NO
+        layoutParams = LayoutParams(LayoutParams.WRAP_CONTENT, 48.dp).apply {
+            gravity = Gravity.CENTER
+        }
+    }
+
+    init {
+        contentDescription = resources.getString(R.string.badge_content_description)
+        importantForAccessibility = View.IMPORTANT_FOR_ACCESSIBILITY_YES
+        layoutParams = LayoutParams(LayoutParams.WRAP_CONTENT, LayoutParams.WRAP_CONTENT)
+        isClickable = true
```

Comment:
I'm not sure the badge is intended to be a pay button. More of a thing to mark a product as eligible for Afterpay

Reply:
Good catch! I meant to take that out. This is still hanging around from when it _was_ a button.